### PR TITLE
Add an AlignTransition widget

### PR DESCRIPTION
**Pull request by amirh**

### CODE REVIEW

File: packages/flutter/lib/src/widgets/transitions.dart [hunk 1]
```diff
@@ -443,6 +443,45 @@ class DecoratedBoxTransition extends AnimatedWidget {
   }
 }
 
+/// Animated version of an [Align] that animates it's [Align.alignment] property.
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/transitions.dart]: `its`

File: packages/flutter/lib/src/widgets/transitions.dart [hunk 2]
```diff
@@ -443,6 +443,45 @@ class DecoratedBoxTransition extends AnimatedWidget {
...
+  ///
+  /// See also:
+  ///
+  /// * [new Align].
+  const AlignTransition({
+    Key key,
+    @required Animation<AlignmentGeometry> alignment,
+    @required this.child,
+    this.widthFactor,
+    this.heightFactor,
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/transitions.dart]: why not make these animated too?

File: packages/flutter/lib/src/widgets/transitions.dart [hunk 3]
```diff
@@ -443,6 +443,45 @@ class DecoratedBoxTransition extends AnimatedWidget {
...
+  }) : super(key: key, listenable: alignment);
+
+  /// The animation that controls the child's alignment.
+  Animation<AlignmentGeometry> get alignment => listenable;
+
+  /// If non-null the child's width factor, see [Align.widthFactor].
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/transitions.dart]: comma after "non-null"

File: packages/flutter/lib/src/widgets/transitions.dart [hunk 4]
```diff
@@ -443,6 +443,45 @@ class DecoratedBoxTransition extends AnimatedWidget {
...
+  final double widthFactor;
+
+  /// If non-null the child's height factor, see [Align.heightFactor].
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/transitions.dart]: ditto

File: packages/flutter/lib/src/widgets/transitions.dart (hunk 2 above)
Comment (Hixie) [reply packages/flutter/lib/src/widgets/transitions.dart]: though i guess it's not clear exactly how that would work...

Comment (Hixie) [reply packages/flutter/lib/src/widgets/transitions.dart]: yeah let's not worry about animating these, and we'll address that problem when it comes around.

**Comment by Hixie** [comment]
LGTM modulo doc fixes
Please make sure Travis and googlebot have run before landing, they weren't running when I reviewed this.

File: packages/flutter/lib/src/widgets/transitions.dart (hunk 2 above)
Comment (abarth) [reply packages/flutter/lib/src/widgets/transitions.dart]: SizeTransition is the dual of this transition that animates those properties.

**Pull request merged**